### ISSUE TITLE
How could we use Leaky ReLU and Parametric ReLU as activation function ?

### ISSUE THREAD

**Issue by gaoyuankidult** [fchollet/keras#117]
Leaky ReLU and Parametric ReLU are implemented as classes, how could we use them as activation functions ?

Thanks

**Comment by patyork** [comment]
There's a PReLU example in the KAggle Otto example; it can be used as a template for all of the Advanced Activation:

```python
from keras.layers.advanced_activations import LeakyReLU
..
..
model.add(LeakyReLU(alpha=.001))
```

**Comment by gaoyuankidult** [comment]
Thanks @patyork . But what if I would like to use them as inner_activation ?

**Comment by patyork** [comment]
Hmm, it doesn't appear that that is possible at the moment. @fchollet may be able to say for sure. This is a good point.

**Comment by fchollet** [comment]
Not currently possible with Keras. It would be possible to modify a recurrent layer to add this capability, though.

**Comment by gaoyuankidult** [comment]
@patyork @fchollet Adding more flags might violate the principles of this library. Perhaps we need to come up with a good way of doing this.

**Comment by wxs** [comment]
@fchollet would a reasonable stopgap approach here be to add a "dummy" layer whose `get_output()` is just the identity, but also exposes the correct PReLU activation as a separate method, say `activation()`? Then by adding it to your model, nothing changes except that its parameters become part of the gradient update.

Finally: when adding the recurrent layer you can set its activation function to `dummy.activation`. Does that make sense? Something like this:

    dummy = DummyPReLU(512)
    model.add(dummy)
    model.add(SimpleRNN(512,512, activation=dummy.activation))

**Comment by kirarpit** [comment]
@iamtodor, I think the problem here is that you didn't define any activation function in the first layer.

**Comment by naarkhoo** [comment]
five years later I have the same question. but seems it is rather a warning now. can you elaborate please.
